go/libraries/doltcore/merge/merge.go

```diff
@@ -170,7 +172,19 @@ func (merger *Merger) MergeTable(ctx context.Context, tblName string) (*doltdb.T
 		return nil, nil, err
 	}
 
-	mergedTable, err := doltdb.NewTable(ctx, merger.vrw, schUnionVal, mergedRowData)
+	emptyIndexData, err := types.NewMap(ctx, merger.vrw)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mergedTable, err := doltdb.NewTable(ctx, merger.vrw, schUnionVal, mergedRowData, emptyIndexData)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	mergedTable, err = mergedTable.RebuildIndexData(ctx)
```

Comment:
This is something we can discuss more, but the current merge path works with tables, of which these aren't anymore. To fix this would be to rewrite the merge logic, and I'm not sure how large of a change it would be, but that's something that I definitely do not plan on doing anytime soon.